### ISSUE TITLE
SC1091 after checking that file exists

### ISSUE THREAD

**Issue by VinzenzBildstein** [koalaman/shellcheck#3000]
#### For bugs
- Rule Id (if any, e.g. SC1000):  SC1091
- My shellcheck version (`shellcheck --version` or "online"):  online
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash

if [ -f /etc/os-release ]; then
  . /etc/os-release
fi

```

#### Here's what shellcheck currently says:

```
$ shellcheck myscript
 
Line 4:
  . /etc/os-release
    ^-- [SC1091](https://www.shellcheck.net/wiki/SC1091) (info): Not following: /etc/os-release was not specified as input (see shellcheck -x).
```

#### Here's what I wanted or expected to see:

Nothing, since the source line only gets reached when /etc/os-release exists.

**Comment by brother** [comment]
if you provide the file on the commandline with --external-sources it will not complain like that.

**Comment by felipecrs** [comment]
@brother can you consider making `/etc/os-release` a default exception for this rule? I believe it's such a common case like:

https://docs.docker.com/engine/install/ubuntu/#install-using-the-repository

```bash
echo \
  "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/docker.asc] https://download.docker.com/linux/ubuntu \
  $(. /etc/os-release && echo "${UBUNTU_CODENAME:-$VERSION_CODENAME}") stable" | \
  sudo tee /etc/apt/sources.list.d/docker.list > /dev/null
```
